egui/src/grid.rs

```diff
@@ -178,15 +178,13 @@ impl GridLayout {
             widget_rect.height().at_least(self.min_cell_size.y),
         );
 
+        cursor.min.x += self.prev_col_width(self.col) + self.spacing.x;
```

Comment:
`frame_rect` here is what is returned by `next_cell`, which currently is the maximum of requested width the the width of the previous column. So afaict the only change here that we advance less than we should in some situations?

Reply:
Opposite, kind of. :) The problem I'm trying to fix here is that it was advancing less than the column width when a cell contains a nested ui.
I'm very new to your code base and this is a bit tricky stuff, but just from looking at where Placer advance_after_rects is called it seems many of those call sites use child_ui.min_rect() to pass as the frame_rect. Which would explain why my fix works. My change makes it advance the full width of the current column, not just the minimum width of the child ui.

Do you think there's something that breaks with this change? Should it be changed to be advance by the maximum of frame_rect width and prev col width?

Reply:
> I'm very new to your code base and this is a bit tricky stuff

I wrote the darn code, and I still don't understand all of it 😅  Doing layout in immediate mode is very tricky, since it is a balancing act of compromises.

> Do you think there's something that breaks with this change? Should it be changed to be advance by the maximum of frame_rect width and prev col width?

Possibly there will be an extra frame of delay when something expands in a grid now, so maybe the maximum would be better. If the grid test would include a slider to change the width of the contents of one cell we could test this.

Reply:
I added a slider now to the table test, tried with and without at least frame rect width. Playing with the slider I was unable to see any difference.